src/officehours_api/models.py

```diff
@@ -96,15 +92,17 @@ class Queue(SafeDeleteModel):
     def hosts_with_phone_numbers(self):
         return get_users_with_emails(self.hosts)
 
+    def remove_allowed_backend(self, old_backend: settings.IMPLEMENTED_BACKEND):
+        new_allowed_backends = list(filter(lambda x: x != old_backend, self.allowed_backends))
+        if len(new_allowed_backends) == 0:
+            new_allowed_backends.append(get_default_backend())
+        self.allowed_backends = new_allowed_backends
+        self.save()
```

Comment:
So you are cool with the change happening here, but you want the `save` to be outside, or do you want the whole operation to not be a method and just be included in the external process? I was somewhat conflicted about whether these two new methods in `models.py` should be on the models at all. Some reading I did (https://medium.com/@jairvercosa/django-model-guideline-d48a96c9b38c) and the extracting of default conditional behavior from the external process made me lean towards this approach.

In Django, bulk operations don't usually require a `save`. I keep wanting to use querysets and bulk operations more for some of this PR, but the reliance on the calculated property `meeting.status` has made it tricky. I see the performance benefit of waiting to `save`, though it's not super relevant with the method uses so far.

Reply:
Oh, I was kinda forgetting about how [`bulk_update`](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#bulk-update) works. Moving `save` out and then doing the bulk update with the modified objects (rather than saving individually) makes a lot of sense.

Reply:
Fat vs. thin models is an eternal question of MVC; in Django I've been leaning toward fat models (not necessarily the right approach, but it feels like an intuitive way to organize things when you're starting out), while in .NET MVC I purposely use thin models and separate my business logic... So I'm inclined to say thin models might often or usually be better, but I'm not sure it would be worth refactoring this project.

And TBH, I didn't remember exactly how `bulk_update` works either, but usually there's a way to "mass save" (commit your transaction), and when it comes to batch jobs, often a record in the middle of your job will blow the whole thing up. In that case, if you were saving individually instead of in bulk at the very end you'll have a mess to clean up.

Reply:
Yeah, it's tricky but interesting to think about; nice to hear your perspective. I think (?) I buy the idea in that article that model methods shouldn't be too dependent on external processes (like the email sending example) and just focus on transforming and modifying data associated with that model. I guess construing "dependency" broadly, you could say referencing `settings` indirectly is a dependency, but that was already something we were doing in `models.py`.

Reply:
K, I think this commit https://github.com/tl-its-umich-edu/remote-office-hours-queue/pull/305/commits/9a001989ad83413f282c2e9ba917e16cbae64884 is what you had in mind. There is only a bulk deletion method for querysets, so I'm not sure I can make that work here for the remaining operation I didn't modify.